src/main/java/ch/njol/skript/expressions/ExprScript.java

```diff
@@ -18,76 +18,108 @@
  */
 package ch.njol.skript.expressions;
 
+import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptCommand;
 import ch.njol.skript.doc.Description;
-import ch.njol.skript.doc.Events;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.parser.ParserInstance;
+import org.jetbrains.annotations.Nullable;
 import org.skriptlang.skript.lang.script.Script;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
 
-@Name("Script Name")
-@Description("Holds the current script's name (the file name without '.sk').")
+import java.io.File;
+import java.util.Arrays;
+import java.util.Objects;
+
+@Name("Script")
+@Description({"The current script, or a script from its (file) name.",
+	"If the script is enabled or disabled (or reloaded) this reference will become invalid.",
+	"Therefore, it is recommended to obtain a script reference <em>when needed</em>."})
```

Comment:
If you disable/reload a script, the data in the old `Script` reference will be discarded. I'd quite like to add some kind of `valid()`/`invalidate()` thing to the Script object to help track that (i.e. is the reference safe to keep?) but it's not essential. Looking for feedback.

Reply:
Added this via a subsequent PR.